cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/RingInstance.java

```diff
@@ -125,40 +126,28 @@ private void writeObject(ObjectOutputStream out) throws IOException
         out.writeUTF(ringEntry.address());
         out.writeInt(ringEntry.port());
         out.writeUTF(ringEntry.datacenter());
-        out.writeUTF(ringEntry.load());
```

Comment:
why do we remove these?

Reply:
Since We are now returning the `StreamResult` back from the tasks, the existing implementation will result in NPEs while serializing the contained `RingInstance`, due to many of these fields not being defined when we create the `RingInstance` from `ReplicaMetadata`.  The change removes the fields not being used from RingInstance from the serialization context.

Likewise, the change is also explicitly setting the `token` field to a default for the same reason, since `token` is part of the equals/hashcode validations.

Stack trace:

```
Caused by: org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 1.0 failed 4 times, most recent failure: Lost task 0.3 in stage 1.0 (TID 40) (172.20.10.7 executor driver): com.esotericsoftware.kryo.KryoException: Error during Java serialization.
Serialization trace:
instance (org.apache.cassandra.spark.bulkwriter.CommitResult)
commitResults (org.apache.cassandra.spark.bulkwriter.StreamResult)
	at org.apache.cassandra.spark.bulkwriter.util.SbwJavaSerializer.write(SbwJavaSerializer.java:58)
	at com.esotericsoftware.kryo.Kryo.writeObject(Kryo.java:575)
```

Reply:
Either that, or if the fields really aren't used, can we simply remove them completely? But, to Francisco's point, we really should handle this cleanly otherwise.

Reply:
Unfortunately, the fields are members of the `RingEntry` instance (part of the sidecar client library) so they cannot be removed here.  

I can add checks for the ones not in `ReplicaMetadata` for completeness, but the intention here was to not write the fields that we know are not used anyway.